jeneamocan/script.rb

```diff
@@ -0,0 +1,204 @@
+require 'pry'
+require 'watir'
+require 'nokogiri'
+require 'open-uri'
+require 'json'
+
+class VB_WebBanking
+  BASE_URL = 'https://web.vb24.md/wb/'.freeze
+  DATA_DIR = 'data'.freeze
+
+  def browser
+    @browser ||= Watir::Browser.new :chrome
+  end
+
+  def run
+    browser.goto(BASE_URL)
+    authentication_check
+  end
+
+  def authentication
+    unless File.exist?('data/login.json')
+    puts "Enter your login"
+    browser.text_field(name: "login").set(gets.chomp)
+    puts "Enter your password"
+    browser.text_field(name: "password").set(gets.chomp)
+    if browser.text_field(name: "captcha").present?
+      puts "Enter CAPTCHA"
+      browser.text_field(name: "captcha").set(gets.chomp)
+      else
+      end
+    else
+      file = File.read('data/login.json')
+      json = JSON.parse(file)
+      login = json["login"]
+      password = json["password"]
+      browser.text_field(name: "login").set(login)
+      browser.text_field(name: "password").set(password)
+    end
+    browser.button(class: "wb-button").click
+    sleep(3)
+    if browser.div(class: "block__cards-accounts").exist?
+      puts "Authentication successful"
+    else
+      puts "Invalid username or password, try again"
+      authentication
+    end
+  end
+
+  def authentication_check
+    if browser.text_field(name: "login").present?
+      puts "Authentication required"
+      authentication
+    else
+    end
+  end
+
+  def info
+    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.NEW_CARDS_ACCOUNTS")
+    name = browser.div(class: "main-info").a(class: "name").text
+    balance = browser.div(class: "primary-balance").span(class: "amount").text
+    currency = browser.div(class: "primary-balance").span(class: %w"amount currency").text
+    nature = browser.div(class: %w"section-title h-small").text.downcase.capitalize
+    puts ["**Account**", "- #{name}", "- #{currency}", "- #{balance}", "- #{nature}"]
+  end
+
+  def parse
+    Nokogiri::HTML.parse(browser.html)
+  end
+
+  def store_result
+    if File.exist?('data/accounts.json') && File.exist?('data/transactions.json')
+      accounts = File.read('data/accounts.json')
+      transactions = File.read('data/transactions.json')
+      accounts_json = JSON.parse(accounts)
+      transactions_json = JSON.parse(transactions)
+      result = accounts_json
+      result["accounts"][0]["transactions"] = transactions_json["transactions"]
+      result
+      Dir.mkdir(DATA_DIR) unless File.exists?(DATA_DIR)
+      file_name = "#{DATA_DIR}/#{File.basename('result')}.json"
+      File.open(file_name, 'w'){|file| file.write(JSON.pretty_generate(result))}
+      puts "Accounts with transactions saved to #{file_name}"
+    else
+      puts "Information is absent"
+    end
+  end
+end
+
+class Accounts < VB_WebBanking
```

Comment:
I'm not sure this is the right way of belonging. Accounts class has nothing to do with parsing script. In the parsing script you should call Account class to create new instance filled with data but not vice-versa